classes/Shortcode.php

```diff
@@ -336,33 +336,42 @@ public function tutor_instructor_list( $atts ) {
 		$current_page = $current_page >= 1 ? $current_page : 1;
 
 		$show_filter         = isset( $atts['filter'] ) ? 'on' === $atts['filter'] : tutor_utils()->get_option( 'instructor_list_show_filter', false );
+		$category_limit      = (int) isset( $atts['category_limit'] ) ? $atts['category_limit'] : 0;
+		$show_empty_category = isset( $atts['show_empty_category'] ) ? '1' === $atts['show_empty_category'] : false;
 		$atts['show_filter'] = $show_filter;
 
 		// Get instructor list to sow.
 		$payload                = $this->prepare_instructor_list( $current_page, $atts );
 		$payload['show_filter'] = $show_filter;
 
+		// empty category filter.
+		$empty_category = 'AND taxonomy.count != 0';
+
 		ob_start();
 		tutor_load_template( 'shortcode.tutor-instructor', $payload );
 		$content = ob_get_clean();
 
+		if ( $category_limit ) {
+			$category_limit = "LIMIT {$category_limit}";
+		} else {
+			$category_limit = '';
+		}
+
+		if ( $show_empty_category ) {
+			$empty_category = '';
+		}
+
 		if ( $show_filter ) {
-			$limit           = 8;
 			$course_taxonomy = CourseModel::COURSE_CATEGORY;
 			$course_cats     = $wpdb->get_results(
 				$wpdb->prepare(
```

Comment:
Refactor the $all_cats & $course_cats using the following way

```
		$term_args = array(
			'taxonomy'   => $course_taxonomy,
			'hide_empty' => ! $show_empty_category,
			'orderby'    => 'count',
			'order'      => 'DESC',
			'number'     => $category_limit ?: 0,
		);

		$course_cats = get_terms( $term_args );

		$all_cats = wp_count_terms( array(
			'taxonomy'   => $course_taxonomy,
			'hide_empty' => false,
		) );
```
After implement test it